source/simulator/core.cc

```diff
@@ -849,17 +851,46 @@ namespace aspect
         {
           Assert (is_element (*p, geometry_model->get_used_boundary_indicators()),
                   ExcInternalError());
-          VectorTools::interpolate_boundary_values (dof_handler,
-                                                    *p,
-                                                    VectorFunctionFromScalarFunctionObject<dim>(std_cxx11::bind (&BoundaryTemperature::Interface<dim>::temperature,
-                                                        std_cxx11::cref(*boundary_temperature),
-                                                        std_cxx11::cref(*geometry_model),
+
+          if (std::isnan(boundary_temperature->boundary_temperature(*p,geometry_model->representative_point(0.0))))
+            {
+              if (timestep_number == 0)
+                pcout << "WARNING: Your selected boundary temperature plugin uses the deprecated "
+                      << "interface function 'temperature(geometry_model,boundary_indicator,position)'. "
+                      << "This function will be removed in a future ASPECT "
+                      << "release. Please update your plugin NOW to the new interface "
+                      << "'boundary_temperature(boundary_indicator,position) "
+                      << "instead, and derive it from SimulatorAccess, if you need access to the "
+                      << "geometry model." << std::endl;
```

Comment:
Sure, it is complicated. 

The problem here is that I want to deprecate the old interface function, to let users know they should update their plugins to the new interface. Usually we do this by deprecating the old function, and implementing a default for the new function that calls the old function, which will throw an exception when both are not implemented in a plugin. In this case however, the new function does not have access to one of the parameters of the old function (the geometry model). So I could only think of one way to do this without breaking every user written plugin: The Simulator needs to check if the new function is implemented, and if it is not, it will output a run-time warning and calls the old function. Therefore I need a default implementation of the new interface function, which returns something that is obviously not the result of a user plugin like a quiet_nan. Using this I can decide within Simulator whether to use the old or new interface. 

So the short answer is: I do not abort, because I want to output a warning for the user, and not throw an error to keep their plugins working the old way for now. It is not pretty to have this additional code in Simulator, but I could not think of another way, and we can remove it after the next release, when users had the chance to update their plugins.